interface.go

```diff
@@ -366,7 +382,9 @@ type RetryHandler interface {
 // RetryHandlerOptions configures the behavior of a RetryHandler.
 type RetryHandlerOptions struct {
 	// MaxRetryAttempts is the maximum number of times that the retry handler is
-	// allowed to attempt to retry a job before it gives up.
+	// allowed to attempt to retry a job before it gives up. This is referring
+	// to the retry handler's attempts to internally retry the job and is
+	// unrelated to the job's particular max attempt setting.
```

Comment:
I am assuming this takes precedence over a job's `MaxAttempts` setting. If so, can we specify that here?

Reply:
I'm not sure what you mean. This is the job-specific max attempts. If you're talking about the retry handler's max attempts, that's referring to the number of times the retry handler is allowed to attempt to re-enqueue the job before it gives up on it and says it's unretryable.

Reply:
Oh, ok I see now. So the job's attempt count will not actually increase until it is enqueued?

Reply:
Yes, this is referring to the `CurrentAttempt` persisted on the job. The retry handler's count just an in-memory count of how many times it's tried to handle the job.